modules/wri_author/src/Commands/WriAuthorCustomCommands.php

```diff
@@ -3,36 +3,213 @@
 namespace Drupal\wri_author\Commands;
 
 use Drush\Commands\DrushCommands;
+use Drupal\wri_author\Entity\WRIAuthor;
+use Drupal\node\Entity\Node;
 
 /**
  * Drush commands for cleaning up wri_author references.
  */
 class WriAuthorCustomCommands extends DrushCommands {
 
   /**
-   * Drush command that displays the given text.
+   * Drush command that consolidates authors by type.
+   *
+   * @param string $type
+   *   Author type - start with internal.
+   * @param int $number
+   *   Number of Authors to process.
+   *
+   * @command wri_author:merge-authors-by-type
+   * @usage wri_author:merge-authors-by-type internal 50
+   */
+  public function mergeAuthorsByType($type = 'internal', $number = 5) {
+    // Find any authors that have the same name, within bundle.
+    // SELECT name FROM wri_author_field_data GROUP BY name HAVING count(name)>1
+    $query = \Drupal::entityQueryAggregate('wri_author')
+      ->groupBy('name')
+      ->condition('type', $type)
+      ->conditionAggregate('name', 'COUNT', '1', '>')
+      ->range(0, $number);
+    $author_list = $query->execute();
+
+    // Get duplicate author IDs by type.
+    foreach ($author_list as $author) {
+      if ($author['name']) {
+        $duplicate_authors = \Drupal::entityQuery('wri_author')
+          ->condition('type', $type)
+          ->condition('name', $author['name'])
+          ->execute();
+      }
+      // Set all references to that author to the first result, i.e.:
+      if ($duplicate_authors) {
+        $first_author_id = array_shift(array_slice($duplicate_authors, 0, 1));
+
+        foreach ($duplicate_authors as $author_id) {
+          // Load all the nodes referencing the bad id.
+          $author_nodes = \Drupal::entityQuery('node')
+            ->condition('field_authors', $author_id, 'IN')
+            ->execute();
+
+          // Switch that bad id out with the good (first) id.
+          if ($author_nodes && ($author_id !== $first_author_id)) {
+            foreach ($author_nodes as $node_id) {
+              $node = Node::load($node_id);
+              $author_list = $node->get('field_authors')->getValue();
+              $key = array_search($author_id, array_column($author_list, 'target_id'));
+              $node->get('field_authors')->removeItem($key);
```

Comment:
I hadn't thought about it initially, but we need to preserve the order of the authors here, since they're attaching them to things like Publications like https://wri.org/research/aqueduct-floods-methodology, and on those the order can matter a lot for citing purposes.

Reply:
It looks like the list of authors is just an array with `target_id` strings as values, so it turned out to be as simple as removing the `removeItem` and adding the `$key` to the array update. The fun part was testing that it worked - 
Before: 
~~~bash
array(5) {
  [0] =>
  array(1) {
    'target_id' =>
    string(4) "8537"
  }
  [1] =>
  array(1) {
    'target_id' =>
    string(5) "20513"
  }
~~~

After: 
~~~bash 
Primary Author: 15682
Duplicate: 20513
Node Updated: 101899

array(5) {
  [0] =>
  array(1) {
    'target_id' =>
    string(4) "8537"
  }
  [1] =>
  array(1) {
    'target_id' =>
    string(5) "15682"
  }
~~~

Double-checked the node in question, and even with a cache clear it keeps the order.